### PR TITLE
fix(proposals): Incremental proposal key for zero proposals (#8005)

### PR DESCRIPTION
**Pull request by anurags92**
Change the proposal's unique key to an atomic counter instead of using a randomly generated key.

(cherry picked from commit a515d0dede9014955f571f96525b49fc8c9913f4) (cherry picked from commit 2aa3d3e5c8f38bfb8c83c4123cdcc86aed58136e)

### CODE REVIEW

**Comment by mangalaman93** [comment]
What's the benefit of this change? Could it cause problems when old Zero's key and new Zero's keys after upgrade collide?

**Review by matthewmcneely — APPROVED** [review]

**Review by mangalaman93 — APPROVED** [review]

**Review by mangalaman93 — DISMISSED** [review]

**Review by matthewmcneely — APPROVED** [review]

**Review by mangalaman93 — APPROVED** [review]

File: worker/proposal.go [hunk 1]
```diff
@@ -23,16 +23,15 @@ import (
 	"sync/atomic"
 	"time"
 
-	"github.com/pkg/errors"
```
Comment (mangalaman93) [inline worker/proposal.go:26]: I think the order is opposite. first external and then dgraph imports.

File: dgraph/cmd/zero/zero_test.go [hunk 2]
```diff
@@ -21,11 +21,13 @@ import (
 	"math"
 	"testing"
 
-	"github.com/stretchr/testify/require"
```
Comment (mangalaman93) [inline dgraph/cmd/zero/zero_test.go:24]: same

**Review by mangalaman93 — CHANGES_REQUESTED** [review]

**Review by mangalaman93 — APPROVED** [review]

**Review by matthewmcneely — DISMISSED** [review]

**Review by mangalaman93 — APPROVED** [review]

File: dgraph/cmd/zero/raft.go [hunk 3]
```diff
@@ -81,8 +84,19 @@ func (n *node) AmLeader() bool {
 	return time.Since(n.lastQuorum) <= 5*time.Second
 }
 
+// {2 bytes Node ID} {4 bytes for random} {2 bytes zero}
+func (n *node) initProposalKey(id uint64) error {
+	x.AssertTrue(id != 0)
+	var err error
+	proposalKey, err = x.ProposalKey(n.Id)
```
[the remark this answers is not in the document]
Comment (anurags92) [reply dgraph/cmd/zero/raft.go:91]: We do need for initialization otherwise we are obscuring that?

**Review by harshil-goel — APPROVED** [review]

**Pull request merged**